frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/index.tsx

```diff
@@ -74,10 +74,12 @@ function GeneButton({
     currentFont
   );
 
-  if (dispatch && length && length > currentMaxLabelHeight) {
```

Comment:
side effect like state change and dispatching events need to be inside a `useEffect()`, otherwise it will cause React error "cannot update component while a child component is being updated`